[Bug-App]: [0.18.0]: Build fails on FreeBSD: undefined: NewGPUNvidia

### Describe the bug

What support for NVidia GPU is WanDB using?
CUDA isn't available on FreeBSD, but Vulkan is available.

@dmitryduev 

When I apply the patch from the PR https://github.com/wandb/wandb/pull/8373 to the release 0.18.0 and add the ```WANDB_BUILD_SKIP_NVIDIA=true``` environment variable the build still fails:
```
Building wandb-core Go binary...
# github.com/wandb/wandb/core/pkg/monitor
pkg/monitor/monitor.go:162:12: undefined: NewGPUAMD
pkg/monitor/monitor.go:171:17: undefined: NewTrainium
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 380, in <module>
    main()
  File "/usr/local/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 364, in main
    json_out["return_val"] = hook(**hook_input["kwargs"])
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/core/pkg/monitor/noop_freebsd.go
+++ b/core/pkg/monitor/noop_freebsd.go
@@ -52,3 +52,60 @@ func (g *GPUNvidia) IsAvailable() bool { return false }
 func (g *GPUNvidia) Probe() *spb.MetadataRequest {
 	return nil
 }
+
+// GPUAMD is a dummy implementation of the Asset interface for AMD GPUs.
+type GPUAMD struct {
+	name   string
+	logger *observability.CoreLogger
+}
+
+func NewGPUAMD(logger *observability.CoreLogger) *GPUAMD {
+	return &GPUAMD{
+		name:   "gpu",
+		logger: logger,
+	}
+}
+
+func (g *GPUAMD) Name() string { return g.name }
+
+func (g *GPUAMD) Sample() (map[string]any, error) { return nil, nil }
+
+func (g *GPUAMD) IsAvailable() bool { return false }
+
+func (g *GPUAMD) Probe() *spb.MetadataRequest {
+	return nil
+}
+
+// Trainium is a dummy implementation of the Asset interface for Trainium.
+type Trainium struct {
+	name                    string
+	pid                     int32
+	samplingInterval        float64
+	logger                  *observability.CoreLogger
+	neuronMonitorConfigPath string
+}
+
+func NewTrainium(
+	logger *observability.CoreLogger,
+	pid int32,
+	samplingInterval float64,
+	neuronMonitorConfigPath string,
+) *Trainium {
+	return &Trainium{
+		name:                    "trainium",
+		pid:                     pid,
+		samplingInterval:        samplingInterval,
+		logger:                  logger,
+		neuronMonitorConfigPath: neuronMonitorConfigPath,
+	}
+}
+
+func (t *Trainium) Name() string { return t.name }
+
+func (t *Trainium) Sample() (map[string]any, error) { return nil, nil }
+
+func (t *Trainium) IsAvailable() bool { return false }
+
+func (t *Trainium) Probe() *spb.MetadataRequest {
+	return nil
+}

